### ISSUE TITLE
Cannot use solidus layout when using Puma workers

### ISSUE THREAD

**Issue by delphaber** [AlchemyCMS/alchemy-solidus#37]
Hi!

I realized that using workers with puma makes alchemy content not to be rendered anymore in solidus layout, even if I followed [this guide](https://github.com/AlchemyCMS/alchemy-solidus#render-alchemy-content-in-solidus-layout).

I created a [repo](https://github.com/delphaber/test-solidus-alchemy) to show you and to propose a fix. Just read the Readme to reproduce the problem :)

I really don't know why this happens, but I tried to use a decorator instead of `require 'alchemy/solidus/use_solidus_layout'` inside the initializers and it seems to work! :)

**Comment by delphaber** [comment]
If you know why it happens, please tell me! I'm curious!

**Comment by tvdeyen** [comment]
Hey, thanks for reporting and the demo application. 

I assume that because of the threaded nature of Puma the initializers won't run again after forking a new thread.

The proposed fix makes sense. Could you provide a PR?

The same should be done in the `lib/alchemy/solidus/alchemy_in_solidus` as well.

Thanks again for reporting 💯

**Comment by delphaber** [comment]
Sure! I can create a PR, next week though!

**Comment by delphaber** [comment]
Actually I have problems with `alchemy_in_solidus` because even if I use the same approach I used for `user_solidus_layout` helpers like `render_navigation` are not present in controllers like Spree::ProductsController, but only in Spree::BaseController. Because something loads `Spree::ProductsController` before this code is excuted:

`Spree::BaseController.send :include, Alchemy::ControllerActions`

Then... I noticed that I have problems with puma workers > 0 only on development. On production it works as expected.

So for my situation I will force puma workers to be `0` in development and I'll keep

```ruby
require 'alchemy/solidus/alchemy_in_solidus'
require 'alchemy/solidus/use_solidus_layout'
```

inside `config/initializers/alchemy.rb`.
